OData.Community.AlternateKeys.V1.md

```diff
@@ -0,0 +1,314 @@
+        <PropertyValue Property="Key">
+          <Collection>
+            <Record>
+              <PropertyValue Property="Name" PropertyPath="ContactInfo/Country" />
+              <PropertyValue Property="Alias" String="Country" />
+            </Record>
+            <Record>
+              <PropertyValue Property="Name" PropertyPath="ContactInfo/Passport" />
+              <PropertyValue Property="Alias" String="Passport" />
+            </Record>
+          </Collection>                
+        </PropertyValue>
+      </Record>
+            
+    </Collection>
+  </Annotation>
+        
+  <Property Name="ID" Type="Edm.Int64" Nullable="false" />
+  <Property Name="SSN" Type="Edm.String" Nullable="true" />
+  <Property Name="ContactInfo" Type="My.ContactInfo" />
+</EntityType>
+```
+
+####2.1.5	Example: Multiple Alternate Keys
+The following metadata described a Person entity type with two alternate key defined.  One key uses the person’s SSN, and the other users their Employee ID.
+
+```xml
+<EntityType Name="Person">
+  <Key>
+    <PropertyRef Name="ID" />
+  </Key>
+        
+  <Annotation Term="Keys.AlternateKeys">
+    <Collection>    
+      <Record>
+        <PropertyValue Property="Key">
+          <Collection>
+            <Record>
+              <PropertyValue Property="Name" PropertyPath="SSN" />
+            </Record>
+          </Collection>
+        </PropertyValue>
+      </Record>
+      <Record>
+        <PropertyValue Property="Key">
+          <Collection>
+            <Record>
+              <PropertyValue Property="Name" PropertyPath="EmployeeID" />
+            </Record>
+          </Collection>
+        </PropertyValue>
+      </Record>
+    </Collection>
+  </Annotation>
+        
+  <Property Name="ID" Type="Edm.Int64" Nullable="false" />
+  <Property Name="SSN" Type="Edm.String" Nullable="true" />
+  <Property Name="EmployeeID" Type="Edm.String" Nullable="true" />
+</EntityType>
+```
+
+####2.1.6	Key Declaration Uniqueness
+An alternate key is identified by the unique set of properties that are referenced in its declaration, regardless of the order in which properties are listed.   
+
+Services SHOULD NOT return multiple alternate key definitions for the same entity type that are composed of the exact same set of properties.
+
+Services MAY define keys that contain overlapping properties or that contain all the properties of another key plus additional properties.
+
+Services SHOULD NOT provide an alternate key definition that exactly matches the property set of the primary entity key.
+ 
+###2.2	Client calls a service with alternate keys
+When a client wishes to address an operation to an entity using alternate key values, they will provide the property name(s) and values for the intended key in the key segment.
+
+####2.2.1	Key Predicate Formats and Key Matching
+OData defines 2 types of key predicates, a simpleKey and a compoundKey as follows:
+
+```http
+http://host/service/persons(123) (simpleKey)
```

Comment:
We initially discusses two "key as segment" flavors: name-value segments
    http://host/service/Customers/CustomerID=ALFKI
and pure-value segments:
    http://host/service/Customers/ALFKI
The latter is the more appealing, but it doesn't allow alternate keys.